metricflow/sql/sql_exprs.py

```diff
@@ -786,6 +791,89 @@ def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D
         return self.sql_function == other.sql_function and self._parents_match(other)
 
 
+class SqlPercentileFunctionType(Enum):
+    """Type of percentile function used."""
+
+    DISCRETE = "discrete"
+    CONTINUOUS = "continuous"
+
+
+@dataclass(frozen=True)
+class SqlPercentileExpressionArgument:
+    """Dataclass for holding percentile arguments"""
+
+    percentile: float
+    function_type: SqlPercentileFunctionType
+
+    @property
+    def function_name(self) -> str:
+        """Helper to build function name to use in rendering"""
+        return "PERCENTILE_CONT" if self.function_type == SqlPercentileFunctionType.CONTINUOUS else "PERCENTILE_DISC"
+
+
+class SqlPercentileExpression(SqlFunctionExpression):
+    """An percentile aggregation expression."""
+
+    def __init__(self, arg: SqlExpressionNode, percentile_args: SqlPercentileExpressionArgument) -> None:
+        """Constructor.
+
+        Args:
+            arg: The expression that should go into the function. e.g. for "percentile_cont(col, 0.1)", the arg
+            expressions should be "col".
+            percentile_args: Auxillary information including percentile value and type.
+        """
+        self.arg = arg
```

Comment:
lets keep it consistent with the other and keep the parameters as a protected attribute in the class and expose it as a property for retrieval. I believe it also helps for type checking.